DataModel/Warning.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+
+namespace ChameleonSystem\UpgradeHelperBundle\DataModel;
+
+use Symfony\Component\Console\Formatter\OutputFormatterStyle;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Warning represents a potential illegal call to the container.
+ */
+final class Warning
+{
+    /** @var string The call is implicit using a variable */
+    public const TYPE_IMPLICIT = 'implicit';
+    /** @var string The call tries to fetch a non existing service */
+    public const TYPE_NON_EXISTENT = 'non-existent';
+
+    /**
+     * @var CallToContainer
+     */
+    public $call;
+    /**
+     * @var string One of self::TYPE_*
+     */
+    public $type;
+
+    /**
+     * format can be used to ouput the warning to the console
+     *
+     * @param OutputInterface $output
+     */
+    public function format(OutputInterface $output)
```

Comment:
Naming: `format` somewhat implies that the entity is formatted and returned in a formatted form instead of being printed to an output.

Structure: While it's not a big problem here, I would probably prefer separating data and output specifics for a command line interface and probably move this method to the command itself.

Reply:
agreed to both. It is now `writeWarning` in the command

Reply:
:ok: